### PR TITLE
binarylog: Use a simple boolean rather than a sync.Once

### PR DESCRIPTION
**Pull request by Jille**
... to start the flusher thread

Given that we pick up the mutex on the line below, we can simply use that for concurrency control.

This removes an atomic read from the hot path.

RELEASE NOTES: N/A

### CODE REVIEW

File: internal/binarylog/sink.go [hunk 1]
```diff
@@ -90,14 +90,17 @@ type bufferedSink struct {
 	out    Sink          // out is built on buf.
 	buf    *bufio.Writer // buf is kept for flush.
 
-	writeStartOnce sync.Once
+	flusherStarted bool
```
Comment (easwars) [inline internal/binarylog/sink.go:93]: Could you please group this new field with the above ones which are currently being protected by the mutex.

**Review by easwars — CHANGES_REQUESTED** [review]

File: internal/binarylog/sink.go (hunk 1 above)
Comment (Jille) [reply internal/binarylog/sink.go:92]: Done

**Comment by Jille** [comment]
Also rebased to master. Apologies if rebasing+squashing isn't desirable.

(Hey Easwar :D)

**Review by easwars — APPROVED** [review]

**Comment by easwars** [comment]
> Also rebased to master. Apologies if rebasing+squashing isn't desirable.

Yes, GitHub can lose track of the relationship between comments and the code location they correspond to, when we force push. But this time it was ok :)

> 
> (Hey Easwar :D)

Hey Jille :)
Is CloudSQL using gRPC now?

**Comment by easwars** [comment]
@Jille 
Sorry, I merged the other PR, so this one has a conflict now.

**Comment by Jille** [comment]
I clicked the "Re-request review" button hoping it'd assign back to you, but apparently it doesn't do that.

Rebased for conflict with my other PR.

**Review by dfawley — APPROVED** [review]

**Pull request merged**